e2e/security/security.go

```diff
@@ -141,23 +147,39 @@ func (c ctx) testSecurityPriv(t *testing.T) {
 		},
 		// capabilities
 		{
-			name:       "capabilities_keep",
-			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
-			opts:       []string{"--keep-privs"},
-			expectExit: 0,
+			// this might break if new capabilities are
+			// added
+			//
+			// TODO(mem): what we really want to test is
+			// that the set outside the container is the
+			// same set inside the container.
+			name:     "capabilities_keep",
+			argv:     []string{"grep", "^CapEff:", "/proc/self/status"},
+			opts:     []string{"--keep-privs"},
+			expectOp: e2e.ExpectOutput(e2e.RegexMatch, `CapEff:\s+0000003fffffffff\n`),
```

Comment:
Old distro like centos 6, sles 11 give `fffffffffffffffff` here. Overall is ping on `127.0.0.1` work ? or running a command like tcpdump to test if capability is dropped ?

Reply:
Pinging 127.0.0.1 instead will work, yes.

But once I started down this path I questioned two things:

- What is this test testing?
- Do I really need to pull 35+ MB of additional containers to test this?

The test as it exists is testing that ping works. It is *assumed* that is happens because the cap_net_raw capability is preserved.

But that's not what --keep-privs is supposed to do. --keep-privs is supposed to preserve _all_ capabilities.

So using ping for this purpose is insufficient, at best. As noted in the comments I added, this new test is brittle. Suggestions as to how to fix this are welcomed.